diff --git a/lib/option.js b/lib/option.js
index 70829540799c0eed6f6e73247ef660208210bd4e..d2cfd908ffb730f68beaa0adffba233455697ba0 100644
--- a/lib/option.js
+++ b/lib/option.js
@@ -315,31 +315,17 @@ function camelcase(str) {
 function splitOptionFlags(flags) {
   let shortFlag;
   let longFlag;
-  // short flag, single dash and single character
-  const shortFlagExp = /^-[^-]$/;
-  // long flag, double dash and at least one character
-  const longFlagExp = /^--[^-]/;
-
-  const flagParts = flags.split(/[ |,]+/).concat('guard');
-  if (shortFlagExp.test(flagParts[0])) shortFlag = flagParts.shift();
-  if (longFlagExp.test(flagParts[0])) longFlag = flagParts.shift();
-
-  // Check for some unsupported flags that people try.
-  if (/^-[^-][^-]/.test(flagParts[0]))
-    throw new Error(
-      `invalid Option flags, short option is dash and single character: '${flags}'`,
-    );
-  if (shortFlag && shortFlagExp.test(flagParts[0]))
-    throw new Error(
-      `invalid Option flags, more than one short flag: '${flags}'`,
-    );
-  if (longFlag && longFlagExp.test(flagParts[0]))
-    throw new Error(
-      `invalid Option flags, more than one long flag: '${flags}'`,
-    );
-  // Generic error if failed to find a flag or an unexpected flag left over.
-  if (!(shortFlag || longFlag) || flagParts[0].startsWith('-'))
-    throw new Error(`invalid Option flags: '${flags}'`);
+  // Use original very loose parsing to maintain backwards compatibility for now,
+  // which allowed for example unintended `-sw, --short-word` [sic].
+  const flagParts = flags.split(/[ |,]+/);
+  if (flagParts.length > 1 && !/^[[<]/.test(flagParts[1]))
+    shortFlag = flagParts.shift();
+  longFlag = flagParts.shift();
+  // Add support for lone short flag without significantly changing parsing!
+  if (!shortFlag && /^-[^-]$/.test(longFlag)) {
+    shortFlag = longFlag;
+    longFlag = undefined;
+  }
 
   return { shortFlag, longFlag };
 }
